File: Java/Bink/src/net/qiujuer/blink/core/Sender.java
package net.qiujuer.blink.core;

/**
 * Send entity interface
 */
public interface Sender {

    /**
     * Send the entity's information
     *
     * @param entity SendEntity
     * @return Status
     */
    boolean sendHead(SendPacket entity);

    /**
     * Send entity
     *
     * @param entity SendEntity
     * @return Status
     */
    boolean sendEntity(SendPacket entity, SendDelivery delivery);

    /**
     * Destroy the sender IO stream
     */
    void destroySendIO();
}

File: C Sharp/Blink/Blink/Core/Sender.cs
using System;
using System.Net.Sockets;

namespace Net.Qiujuer.Blink.Core
{
    public interface Sender : IDisposable
    {
        /// <summary>
        /// Get socket send buffer size
        /// </summary>
        /// <returns>Buffer Size</returns>
        int GetBufferSize();

        /// <summary>
        /// Async send some byte
        /// </summary>
        /// <param name="e">SocketAsyncEventArgs</param>
        /// <returns>Status</returns>
        bool SendAsync(SocketAsyncEventArgs e);
    }
}
